### DISCUSSION TITLE
DataGrid Multiple Selection

### DISCUSSION THREAD

**Discussion by meekhasaji** [Megabit/Blazorise#2782] [Q&A] [answered]
Currently , I have a datagrid  with 

 ```
<DataGrid @ref="ModelDataGrid"
                                      TItem="Model"
                                      Data="@_modellist"
                                      Editable="@CanEdit()"
                                      Sortable="true"
                                      SortMode="DataGridSortMode.Multiple"
                                      ShowPager="true"
                                      PagerPosition="DataGridPagerPosition.Bottom"
                                      ShowPageSizes="true"
                                      PageSize="10"
                                      Striped="true"
                                      Bordered="true"
                                      Hoverable="true"
                                      RowSelectable="_ => true"
                                      Responsive="true"
                                      Filterable="false"
                                      CustomFilter="@OnCustomFilter"
                                      @bind-SelectedRow="@_selectedModel"
                                      RowClicked="row => SetToggle(_isExpanded)"
                                      SelectedRowStyling="@OnSelectedRowStyling"
                                      DetailRowTrigger="@((item)=>(item.Id == _selectedModel?.Id && _isExpanded))">
```

what can we do inorder expand multiple rows at the same time .

Also, sometimes the data that I have binded doesn't come up, the grid shows empty values and loads the grid, what can we do to resolve this.

**Comment by David-Moreira** [comment, marked as answer]
Hello @meekhasaji 
Once you set the Datagrid to multiple selection, you should now use the `SelectedRows` parameter to track which rows have been selected.
https://blazorise.com/docs/extensions/datagrid/#datagrid-multiple-selection
With this in mind, you should be able to expand multiplerows by now using the `DetailRowTrigger` to check if the item is in the `SelectedRows`. 
Should be something like: 
`DetailRowTrigger="@((item)=>( _selectedRules?.Any(x=> x.Id == item.Id) && _isExpanded))"`


------------
> Also, sometimes the data that I have binded doesn't come up, the grid shows empty values and loads the grid, what can we do to resolve this.

Could you elaborate on this problem? How are you loading your `_rules` collection? And are you making changes to it?

**Reply by meekhasaji** [reply]
@David-Moreira ,thank you for replying, I'll try the first suggestion out.
the data grid loads with empty cell values sometimes, no changes are being made, when I click a button to load the rules to the datagrid

**Reply by David-Moreira** [reply]
Is the call asynchronous? From DB? A repository replicating the problem makes it easier to analyse why sometimes it displays empty, if that's possible. Thanks.

**Reply by meekhasaji** [reply]
Hi @David-Moreira , thanks it got resolved, changed it to a synchronous call. I'll upload a POC replicating this earlier scenario shortly.
Thanks again!
